import wollok.lang.Exception

class AException inherits Exception { }
class BException inherits Exception { }
			
class A { 
	method m1() { throw new AException(message = "hello you see") }
}
			
test "Test catch without type matching just the first catch" {
	
	const a = new A()
				
	try {
		a.m1()
		assert.fail("Should have thrown exception")
	}	
				
	catch e : AException {
		// OK !
	}

	catch e : BException {
		assert.fail("incorrenct catch !")
	}
	
	catch e {
		assert.fail("incorrenct catch !")
	}

}